### ISSUE TITLE
Redundant if test lint

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust-clippy#8923]
### What it does

Here the first if test contains a return so the second if can't be true. This is usually a coding mistake that could hide a bug, so I think a lint that spots such situations is good. This lint could be complex (like firing even if the first test contains "< 3") but even basic cases like this with repeated conditions should be useful.

### Lint Name

impossibile_if_branch

### Category

suspicious

### Example

```rust
fn test2(s: &[u32]) -> bool {
    if s.len() < 2 {
        return true;
    }
    
    if s.len() < 2 {
        println!("*");
    }
    false
}
```
